pkg/api/validation/validation.go

```diff
@@ -1237,6 +1256,21 @@ func ValidatePersistentVolume(pv *api.PersistentVolume) field.ErrorList {
 			allErrs = append(allErrs, validateScaleIOVolumeSource(pv.Spec.ScaleIO, specPath.Child("scaleIO"))...)
 		}
 	}
+	if pv.Spec.LocalVolume != nil {
+		if numVolumes > 0 {
+			allErrs = append(allErrs, field.Forbidden(specPath.Child("localVolume"), "may not specify more than 1 volume type"))
+		} else {
+			numVolumes++
+			if !utilfeature.DefaultFeatureGate.Enabled(features.PersistentLocalVolumes) {
+				allErrs = append(allErrs, field.Forbidden(specPath.Child("localVolume"), "LocalVolumes are disabled by feature-gate"))
+			}
+			allErrs = append(allErrs, validateLocalVolumeSource(pv.Spec.LocalVolume, specPath.Child("localVolume"))...)
+			// StorageClass is required
+			if len(pv.Spec.StorageClassName) == 0 {
```

Comment:
> We could put the topologyKey in the PVC instead, but I prefer the idea of the StorageClass encapsulating the topology constraints so that the user does not need to be aware.

A user can still manually bind local PVs to PVCs without storage classes. Why force every local PV to expose a storage class?
I'm personally not a fan of tying storage classes to PVs during static provisioning.
In the future, will it be possible to keep these operations separate?
1. Provisioning local volumes
2. Defining storage classes for grouping local volumes